main.go

```diff
@@ -53,8 +54,10 @@ func loadConfig() (*config.Configuration, error) {
 func serve(revision string, cfg *config.Configuration) error {
 	fetchingInterval := time.Duration(cfg.CurrencyConverter.FetchIntervalSeconds) * time.Second
 	staleRatesThreshold := time.Duration(cfg.CurrencyConverter.StaleRatesSeconds) * time.Second
-	currencyConverter := currencies.NewRateConverter(&http.Client{}, cfg.CurrencyConverter.FetchURL,
-		fetchingInterval, staleRatesThreshold)
+	currencyConverter := currencies.NewRateConverter(&http.Client{}, cfg.CurrencyConverter.FetchURL, fetchingInterval, staleRatesThreshold)
+	
+	currencyConverterTickerTask := task.NewTickerTask(fetchingInterval, currencyConverter)
+	currencyConverterTickerTask.Start()
```

Comment:
I'm not sure if its necessary but, should we also add `defer currencyConverterTickerTask.Stop()`?

Reply:
I don't think it's necessary; `Stop()` is just closing a channel which halts the timer. I guess I don't see any risk in an extra tick or two occurring while in the process of a shutdown, especially since the ticker is just being used for rate conversions at this time.

From the golang docs: "Channels aren't like files; you don't usually need to close them. Closing is only necessary when the receiver must be told there are no more values coming, such as to terminate a range loop."

Also deferred calls don't run when the main go routine exits. I think we would need to use sync waitgroups for this sort of thing.